src/index/Index.cpp

```diff
@@ -26,12 +26,8 @@ void Index::createFromOnDiskIndex(const std::string& onDiskBase,
 }
 
 // ____________________________________________________________________________
-void Index::buildTextIndexFile(
-    const std::optional<std::pair<std::string, std::string>>& wordsAndDocsFile,
-    bool addWordsFromLiterals, TextScoringMetric textScoringMetric,
-    std::pair<float, float> bAndKForBM25) {
-  pimpl_->buildTextIndexFile(wordsAndDocsFile, addWordsFromLiterals,
-                             textScoringMetric, bAndKForBM25);
+void Index::buildTextIndexFile(TextIndexConfig config) {
+  pimpl_->buildTextIndexFile(config);
```

Comment:
I'm sure sonar wants to `std::move` here or use a `const&` whatever you need in the impl.